### PR TITLE
723 Update science metadata from REST API

### PR DESCRIPTION
**Pull request by selimnairb**
This is a work in progress.  Updating resource for GenericResource has rudimentary tests.  Need to write tests for derived classes, starting with SWAT model instance.

### CODE REVIEW

**Comment by hydrobuild** [comment]
Can one of the admins verify this patch?

File: hs_core/views/resource_rest_api.py [hunk 1]
```diff
@@ -475,12 +485,75 @@ def get(self, request, pk):
...
+        expect = 'resourcemetadata.xml'
+        if scimeta.name != expect:
+            error_msg = {'file': "Uploaded file has name {n}, but expected {e}.".format(n=scimeta.name,
+                                                                                        e=expect)}
+            raise ValidationError(detail=error_msg)
+
+        # Temp directory to store resourcemetadata.xml
+        tmp_dir = tempfile.mkdtemp()
+        try:
+            # Fake the bag structure so that GenericResourceMeta.read_metadata_from_resource_bag
```
Comment (selimnairb) [inline hs_core/views/resource_rest_api.py]: Probably want to move some of this into the serialization API.

**Comment by jbradberry** [comment]
Aside from a couple of minor tweaks (which I've pushed), I'm good with this.

**Comment by pkdash** [comment]
I only reviewed @jbradberry changes and +1 for those changes.

**Comment by pkdash** [comment]
@mjstealey and @jbradberry  I assume this is ready to be merged.

**Comment by jbradberry** [comment]
@pkdash I would say so

**Pull request merged**